### PR TITLE
cpr: new package

### PR DESCRIPTION
**Pull request by sethrj**
Dependency of [XACC](https://github.com/eclipse/xacc).

### CODE REVIEW

File: var/spack/repos/builtin/packages/cpr/package.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        if self.version < Version("1.10"):
+            _force = "_FORCE"
+        else:
+            _force = ""
+        args = [
+            self.define("CPR_USE_SYSTEM_GTEST", True),
+            self.define(f"CPR{_force}_USE_SYSTEM_CURL", True),
+            self.define("CPR_ENABLE_SSL", True),
+        ]
+        return args
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/cpr/package.py]:
```suggestion
        _force = "_FORCE" if self.spec.satisfies("@:1.9") else ""
        
        return [
            self.define("CPR_USE_SYSTEM_GTEST", True),
            self.define(f"CPR{_force}_USE_SYSTEM_CURL", True),
            self.define("CPR_ENABLE_SSL", True),
        ]

```

We can use `satisfies()` here to get around having to create a new version object which cleans up the syntax a bit.

**Review by alecbcs — CHANGES_REQUESTED** [review]
@sethrj this PR looks really good! Just a quick suggestion to clean up the args creation a bit.

**Comment by alecbcs** [comment]
@spackbot fix style

**Comment by sethrj** [comment]
@alecbcs requires reapproval

**Review by alecbcs — APPROVED** [review]
Confirmed SHAs. @sethrj thanks for adding this new package!

**Pull request merged**